.buildkite/steps/tests.sh

```diff
@@ -1,13 +1,25 @@
 #!/bin/bash
-set -euo pipefail
 
 go version
 echo arch is "$(uname -m)"
 
 go install gotest.tools/gotestsum@v1.8.0
 
 echo '+++ Running tests'
-gotestsum --junitfile "junit-${OSTYPE}.xml" -- -count=1 -failfast "$@" ./...
+gotestsum --junitfile "junit-${OSTYPE}.xml" -- -count=1 "$@" ./... || TEST_EXIT_STATUS=$?
 
 echo '+++ Running integration tests for git-mirrors experiment'
-TEST_EXPERIMENT=git-mirrors gotestsum --junitfile "junit-${OSTYPE}-git-mirrors.xml" -- -count=1 -failfast "$@" ./bootstrap/integration
+TEST_EXPERIMENT=git-mirrors gotestsum --junitfile "junit-${OSTYPE}-git-mirrors.xml" -- -count=1 "$@" ./bootstrap/integration || TEST_EXIT_STATUS_WITH_MIRROS=$?
+
+echo '+++ Test Analytics'
+if [[ -n "${TEST_ANALYTICS_TOKEN_ENV_KEY-}" ]]; then
+  cat "junit-${OSTYPE}.xml" | TEST_ANALYTICS_TOKEN=${!TEST_ANALYTICS_TOKEN_ENV_KEY} bash -c "`curl -sL https://raw.githubusercontent.com/buildkite/collector-junit/main/collect.sh`"
+else
+  echo "No TEST_ANALYTICS_TOKEN_ENV_KEY is present, skipping..."
+fi
+
+if [[ $TEST_EXIT_STATUS -ne 0 ]]; then
+  exit $TEST_EXIT_STATUS
+elif [[ $TEST_EXIT_STATUS_WITH_MIRROS -ne 0 ]]; then
+  exit $TEST_EXIT_STATUS_WITH_MIRROS
+fi
```

Comment:
Instead of making this tests.sh file more complicated, could we make the test analytics collection a step in its own right? This script is dead simple at the moment, and i'd really like to keep it that way

Reply:
Happy to make that change, how do you propose I do that? Running the tests and uploading the junit xml is kinda linked at the hip.

Reply:
We currently have another step for annotating the build using the JUnit files - could we do something similar here? add a step that has `artifact-paths: junit-*.xml` and upload them from there?